src/profiles.rs

```diff
@@ -84,6 +84,14 @@ impl Profiles {
             None
         }
     }
+
+    pub fn iter(&self) -> impl Iterator<Item = &Arc<Profile>> {
+        self.trusted
+            .iter()
+            .chain(self.pool.iter())
+            .chain(self.dirty.iter())
+            .map(|(_k, v)| v)
+    }
```

Comment:
what about instead we give access to the different sets and let the user chain them the way they want when accessing the profiles ?